### PR TITLE
[beta] Fixes for 12.2

### PR DESCRIPTION
**Pull request by tudorms**
Populate the beta branch with the Beta version of V8 (12.2). This version included additional checks for isolate entrance/exit consistency which we were failing before, because apparently we kept the Isolated "entered" all the time on the thread.

This change implies that we may create more than one Isolate (even on the same thread) so that's something to look out for (potential perf impact).
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/v8-jsi/pull/189)

### CODE REVIEW

**Review by Khalef1 — APPROVED** [review]

File: config.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "version":  "0.71.13",
-    "v8ref":  "refs/branch-heads/12.1",
-    "buildNumber":  "285"
+    "version":  "0.0.1",
```
Comment (vmoroz) [inline config.json:2]: Is it intentional that the version is changed to 0.0.1?

File: src/node-api-jsi/NodeApiJsiRuntime.cpp [hunk 2]
```diff
@@ -1528,15 +1528,13 @@ jsi::Value NodeApiJsiRuntime::callAsConstructor(const jsi::Function &func, const
 }
 
 jsi::Runtime::ScopeState *NodeApiJsiRuntime::pushScope() {
-  NodeApiEnvScope scope{getEnv()};
```
Comment (vmoroz) [inline src/node-api-jsi/NodeApiJsiRuntime.cpp:1531]: Do we pass all unit tests after this change?

Comment (tudorms) [reply src/node-api-jsi/NodeApiJsiRuntime.cpp:1531]: Yes. Seems like this was mismatched. enter1 -> enter2 -> exit1 -> exit2, and was crashing in V8 code.

File: config.json (hunk 1 above)
Comment (tudorms) [reply config.json:2]: Yes, just in case this accidentally beta bits end up in the release pipeline, want to be able to spot it easily.

**Pull request merged**